query-engine/core/src/query_document/parser.rs

```diff
@@ -95,8 +96,8 @@ impl QueryDocumentParser {
         schema_field: &OutputFieldRef,
         given_arguments: &[(String, ArgumentValue)],
     ) -> QueryParserResult<Vec<ParsedArgument>> {
-        let valid_argument_names: HashSet<&str> = schema_field.arguments.iter().map(|arg| arg.name.as_str()).collect();
-        let given_argument_names: HashSet<&str> = given_arguments.iter().map(|arg| arg.0.as_str()).collect();
+        let valid_argument_names: IndexSet<&str> = schema_field.arguments.iter().map(|arg| arg.name.as_str()).collect();
+        let given_argument_names: IndexSet<&str> = given_arguments.iter().map(|arg| arg.0.as_str()).collect();
```

Comment:
This is because the order of fields in the error messages are unstable with HashSet. That makes the tests not deterministic.